### PR TITLE
Fix api version for cronjobs to display with dashboard module

### PR DESCRIPTION
**Pull request by Beastlex**
## Description

Fix api version for cronjobs to display with dashboard module

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dashboard
type: fix
summary: Fix apiVersion for CronJobs to display with the dashboard module.
impact_level: default
```

### CODE REVIEW

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 1]
```diff
@@ -2,12 +2,27 @@
 ARG BASE_ALPINE
 FROM kubernetesui/dashboard:v2.7.0@sha256:2e500d29e9d5f4a086b908eb8dfe7ecac57d2ab09d65b24f588b1d449841ef93 as artifact
 
+# to fix dashboard can display cronjobs v1
+FROM golang:1.19.12-alpine3.18 as patchbuilder
```
Comment (aleksey-su) [inline modules/500-dashboard/images/dashboard/Dockerfile:6]:
```suggestion
ARG BASE_GOLANG_19_ALPINE

# to fix dashboard can display cronjobs v1
FROM $BASE_GOLANG_19_ALPINE as patchbuilder
```

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 2]
```diff
@@ -2,12 +2,27 @@
...
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -ldflags "-X github.com/kubernetes/dashboard/src/app/backend/client.Version=v2.7.0" -o dashboard .
+
+
 FROM $BASE_ALPINE
 
 COPY --from=artifact /etc/passwd /etc/passwd
 COPY --from=artifact /public /public
 COPY --from=artifact /locale_conf.json /locale_conf.json
-COPY --from=artifact /dashboard /dashboard
+# COPY --from=artifact /dashboard /dashboard
```
Comment (aleksey-su) [inline modules/500-dashboard/images/dashboard/Dockerfile:24]:
```suggestion
```

**Review by aleksey-su — CHANGES_REQUESTED** [review]
Please create a file README.md in the patches folder and describe what your patch does

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 3]
```diff
@@ -2,12 +2,27 @@
 ARG BASE_ALPINE
 FROM kubernetesui/dashboard:v2.7.0@sha256:2e500d29e9d5f4a086b908eb8dfe7ecac57d2ab09d65b24f588b1d449841ef93 as artifact
 
+# to fix dashboard can display cronjobs v1
+FROM golang:1.19.12-alpine3.18 as patchbuilder
+
+WORKDIR /src/
+RUN apk add --no-cache git patch make zip wget && \
+    wget -c https://github.com/kubernetes/dashboard/archive/refs/tags/v2.7.0.zip && \
```
Comment (RomanenkoDenys) [inline modules/500-dashboard/images/dashboard/Dockerfile:10]: use git clone here

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 4]
```diff
@@ -2,12 +2,27 @@
...
+    unzip v2.7.0.zip
+COPY patches/001-fix-cronjob-ver.patch /src/
+RUN patch -u dashboard-2.7.0/src/app/backend/resource/cronjob/jobs.go -i 001-fix-cronjob-ver.patch --ignore-whitespace --fuzz 3 && \
```
Comment (RomanenkoDenys) [inline modules/500-dashboard/images/dashboard/Dockerfile:13]: use git apply here

**Comment by aleksey-su** [comment]
@Beastlex I suggest using cherry-pick from commit 'f79f3a5e25df152df6164ddfaf2ffc1f09e5058f' instead of using patch

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Comment by Beastlex** [comment]
Create test CronJob with content:

**Comment by Beastlex** [comment]
```
apiVersion: batch/v1
kind: CronJob
metadata:
  name: hello
spec:
  schedule: "10 * * * *"
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - name: hello
            image: busybox:1.28
            imagePullPolicy: IfNotPresent
            command:
            - /bin/sh
            - -c
            - date; echo Hello from the Kubernetes cluster
          restartPolicy: OnFailure
```

**Comment by Beastlex** [comment]
image

**Comment by Beastlex** [comment]
image

**Review by aleksey-su — APPROVED** [review]

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 5]
```diff
@@ -1,13 +1,29 @@
 # Based on https://github.com/kubernetes/dashboard/blob/v2.7.0/aio/Dockerfile
 ARG BASE_ALPINE
+ARG BASE_GOLANG_19_ALPINE
 FROM kubernetesui/dashboard:v2.7.0@sha256:2e500d29e9d5f4a086b908eb8dfe7ecac57d2ab09d65b24f588b1d449841ef93 as artifact
 
+# to fix dashboard can display cronjobs v1
+FROM $BASE_GOLANG_19_ALPINE as patchbuilder
+
+WORKDIR /src/
+RUN apk add --no-cache git patch make zip wget && \
```
Comment (RomanenkoDenys) [inline modules/500-dashboard/images/dashboard/Dockerfile:10]:
```suggestion
RUN apk add --no-cache git patch make zip && \
```

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 6]
```diff
@@ -1,13 +1,29 @@
...
+    git clone https://github.com/kubernetes/dashboard && \
```
Comment (RomanenkoDenys) [inline modules/500-dashboard/images/dashboard/Dockerfile:11]:
```suggestion
    git clone ${SOURCE_REPO}/kubernetes/dashboard && \
```

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 7]
```diff
@@ -1,13 +1,29 @@
...
+    cd dashboard && \
+    git config --global user.email "builder@deckhouse.io" && \
+    git checkout v2.7.0 && \
+    git cherry-pick f79f3a5e25df152df6164ddfaf2ffc1f09e5058f && \
+    go mod download
+WORKDIR /src/dashboard/src/app/backend/
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -ldflags "-X github.com/kubernetes/dashboard/src/app/backend/client.Version=v2.7.0" -o dashboard .
```
Comment (RomanenkoDenys) [inline modules/500-dashboard/images/dashboard/Dockerfile:18]:
```suggestion
RUN go build -a -installsuffix cgo -ldflags "-X github.com/kubernetes/dashboard/src/app/backend/client.Version=v2.7.0" -o dashboard .
```

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 8]
```diff
@@ -1,13 +1,29 @@
 # Based on https://github.com/kubernetes/dashboard/blob/v2.7.0/aio/Dockerfile
 ARG BASE_ALPINE
+ARG BASE_GOLANG_19_ALPINE
 FROM kubernetesui/dashboard:v2.7.0@sha256:2e500d29e9d5f4a086b908eb8dfe7ecac57d2ab09d65b24f588b1d449841ef93 as artifact
 
+# to fix dashboard can display cronjobs v1
+FROM $BASE_GOLANG_19_ALPINE as patchbuilder
```
Comment (RomanenkoDenys) [inline modules/500-dashboard/images/dashboard/Dockerfile:7]:
```suggestion
FROM $BASE_GOLANG_19_ALPINE as patchbuilder

ARG GOPROXY
ARG SOURCE_REPO

ENV GOPROXY=${GOPROXY} \
    SOURCE_REPO=${SOURCE_REPO} \
    CGO_ENABLED=0 \
    GOOS=linux \
    GOARCH=amd64
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
Some nits

**Review by nabokihms — APPROVED** [review]

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 9]
```diff
@@ -1,13 +1,38 @@
...
+
+ENV GOPROXY=${GOPROXY} \
+    SOURCE_REPO=${SOURCE_REPO} \
+    CGO_ENABLED=0 \
+    GOOS=linux \
+    GOARCH=amd64
+
+WORKDIR /src/
+RUN apk add --no-cache git patch make zip && \
+    git clone https://${SOURCE_REPO}/kubernetes/dashboard && \
```
Comment (RomanenkoDenys) [inline modules/500-dashboard/images/dashboard/Dockerfile:20]:
```suggestion
    git clone ${SOURCE_REPO}/kubernetes/dashboard && \
```

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 10]
```diff
@@ -1,13 +1,38 @@
...
+    GOOS=linux \
+    GOARCH=amd64
+
+WORKDIR /src/
+RUN apk add --no-cache git patch make zip && \
+    git clone ${SOURCE_REPO}/kubernetes/dashboard && \
+    cd dashboard && \
+    git config --global user.email "builder@deckhouse.io" && \
+    git checkout v2.7.0 && \
+    git cherry-pick f79f3a5e25df152df6164ddfaf2ffc1f09e5058f && \
```
Comment (RomanenkoDenys) [inline modules/500-dashboard/images/dashboard/Dockerfile:24]:
```suggestion
    # Use batch/v1 for fetching cronjob jobs https://github.com/kubernetes/dashboard/pull/7465
    git cherry-pick f79f3a5e25df152df6164ddfaf2ffc1f09e5058f && \
```

File: modules/500-dashboard/images/dashboard/Dockerfile [hunk 11]
```diff
@@ -1,13 +1,39 @@
 # Based on https://github.com/kubernetes/dashboard/blob/v2.7.0/aio/Dockerfile
 ARG BASE_ALPINE
+ARG BASE_GOLANG_19_ALPINE
 FROM kubernetesui/dashboard:v2.7.0@sha256:2e500d29e9d5f4a086b908eb8dfe7ecac57d2ab09d65b24f588b1d449841ef93 as artifact
```
Comment (aleksey-su) [inline modules/500-dashboard/images/dashboard/Dockerfile:4]:
```suggestion
```

**Review by aleksey-su — CHANGES_REQUESTED** [review]

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Pull request merged**